docs/proposals/202003_thanos_rules_federation.md

```diff
@@ -0,0 +1,81 @@
+---
+title: Global / Federated Rules API
+type: proposal
+menu: proposals
+status: accepted
+owner: s-urbaniak,bwplotka
+---
+
+### Related Tickets
+
+https://github.com/thanos-io/thanos/issues/1375
+
+## Summary
+
+Thanos allows a global query view for the Prometheus series. This means discovering, connecting various (often remote) "leafs" components and aggregating series data from them.
+
+Since we work with metrics, evaluating recording rules and alerts is a very important part of our system. Some of our "StoreAPIs" like Prometheus and Thanos Ruler are designed for that. However, we currently don’t have a way to present those resources in a federated way e.g Alerts and Recording Rules. This document explores the potential way of solving this in the Prometheus Ecosystem.
+
+## Motivation
+
+Deployment scenarios with various levels of federated setups are not unusual. Concrete deployments i.e. leverage Thanos Ruler to spawn multiple query sources (i.e. via Thanos Querier) for rule and alert evaluation. This topology sacrifices availability. At the same time in the same deployment topology a dedicated Prometheus instance is deployed for certain critical alerting rules implying high availability. Here, we are missing a federation middleware exposing a consolidated view of evaluated rules and alerts in the underlying Prometheus and ThanosRuler instances.
+
+Pitfalls of current solutions:
+* It’s tedious to manually visit each leaf and we are lazy.
+* Visiting leaf instances (be it ThanosRuler or dedicated Prometheus endpoints) have to be implemented manually.
+
+## Goals
+
+* Present other Prometheus/Thanos Resources in a global view.
+    * This also means an up-to-date view (e.g statuses)
+* It can be tied to Thanos.
```

Comment:
Not sure what do you mean here?

Reply:
I am not sure as well ;p 

I think it can be "opinionated" and tailored for Thanos needs? I agree with @povilasv that it sounds weird as it is Thanos proposal after all...

Reply:
Let's remove maybe

Reply:
agreed, removed